cpp/src/core/sroptcryst.h

```diff
@@ -948,60 +950,51 @@ class srTOptCryst : public srTGenOptElem {
 		complex<double> x2C = (-zeeC - sqrqzC) / m_psimhc;
 		//complex<double> ph1C = 2.*pi*iC*k0Ai*del1C*(m_thicum*1.E+04)/gamma0;
 		//complex<double> ph2C = 2.*pi*iC*k0Ai*del2C*(m_thicum*1.E+04)/gamma0;
-		complex<double> aux_phC = 2.*pi*iC*k0Ai*(m_thicum*1.E+04) / gamma0;
-		complex<double> ph1C = aux_phC*del1C;
-		complex<double> ph2C = aux_phC*del2C;
-
-		complex<double> DHsgC, Cph1C, Cph2C, D0trsC;
-		if(real(ph1C) > logDBMax) DHsgC = x2C;
-		else if(real(ph2C) > logDBMax) DHsgC = x1C;
-		else
+		complex<double> aux_phC = 2. * pi * iC * k0Ai * (m_thicum * 1.E+04) / gamma0; //m_thicum goes to A
+		complex<double> ph1C = aux_phC * del1C;
+		complex<double> ph2C = aux_phC * del2C;
+		complex<double> Cph1C, Cph2C;
+		Cph1C = exp(ph1C);
+		Cph2C = exp(ph2C);
```

Comment:
Left the previous version. The previous John's code, that is suggested to be removed here, seems to make some numerical overflow protection, and it refers to the Bragg case (that is not the subject of this update).